x-pack/plugins/apm/server/index.ts

```diff
@@ -26,6 +26,8 @@ const configSchema = schema.object({
   serviceMapTraceIdBucketSize: schema.number({ defaultValue: 65 }),
   serviceMapTraceIdGlobalBucketSize: schema.number({ defaultValue: 6 }),
   serviceMapMaxTracesPerRequest: schema.number({ defaultValue: 50 }),
+  serviceMapTerminateAfter: schema.number({ defaultValue: 100_000 }),
+  serviceMapMaxTraces: schema.number({ defaultValue: 1000 }),
```

Comment:
What about a setting like `serviceMapMaxItemsPerTraces` to prevent large traces (10.000+ items) to consume all memory? Without trace routing we can only do this on the coordinating node but still better than nothing.

Reply:
I meant inside the scripted metric agg. When stitching the traces together, we can count the number of items in a trace. If we reach X we discard it (or something like that).